encodings/runend/src/compress.rs

```diff
@@ -16,27 +17,56 @@ pub fn runend_encode(array: &PrimitiveArray) -> (PrimitiveArray, PrimitiveArray)
 
-    match_each_native_ptype!(array.ptype(), |$P| {
+    let (compressed_ends, compressed_values) = match_each_native_ptype!(array.ptype(), |$P| {
         let (ends, values) = runend_encode_primitive(array.maybe_null_slice::<$P>());
+        (PrimitiveArray::from_vec(ends, Validity::NonNullable), PrimitiveArray::from_vec(values, validity))
+    });
 
-        let mut compressed_values = PrimitiveArray::from_vec(values, validity);
-        compressed_values.statistics().set(Stat::IsConstant, false.into());
-        compressed_values.statistics().set(Stat::RunCount, compressed_values.len().into());
-        array.statistics().get(Stat::Min).map(|s| compressed_values.statistics().set(Stat::Min, s));
-        array.statistics().get(Stat::Max).map(|s| compressed_values.statistics().set(Stat::Max, s));
-        array.statistics().get(Stat::IsSorted).map(|s| compressed_values.statistics().set(Stat::IsSorted, s));
-        array.statistics().get(Stat::IsStrictSorted).map(|s| compressed_values.statistics().set(Stat::IsStrictSorted, s));
-
-        let compressed_ends = PrimitiveArray::from(ends);
-        compressed_ends.statistics().set(Stat::IsSorted, true.into());
-        compressed_ends.statistics().set(Stat::IsStrictSorted, true.into());
-        compressed_ends.statistics().set(Stat::IsConstant, false.into());
-        compressed_ends.statistics().set(Stat::Max, array.len().into());
-        compressed_ends.statistics().set(Stat::RunCount, compressed_ends.len().into());
-
-        assert_eq!(array.dtype(), compressed_values.dtype());
-        (compressed_ends, compressed_values)
-    })
+    // the values array stats are trivially derived
+    compressed_values
+        .statistics()
+        .set(Stat::RunCount, compressed_values.len().into());
+    compressed_values
+        .statistics()
+        .set(Stat::IsConstant, (compressed_values.len() == 1).into());
+    if let Some(min) = array.statistics().get(Stat::Min) {
+        compressed_values.statistics().set(Stat::Min, min);
+    }
+    if let Some(max) = array.statistics().get(Stat::Max) {
+        compressed_values.statistics().set(Stat::Max, max);
+    }
+    if let Some(is_sorted) = array.statistics().get(Stat::IsSorted) {
+        compressed_values
+            .statistics()
+            .set(Stat::IsSorted, is_sorted);
+    }
+    if let Some(is_strict_sorted) = array.statistics().get(Stat::IsStrictSorted) {
+        compressed_values
+            .statistics()
+            .set(Stat::IsStrictSorted, is_strict_sorted);
+    }
+
+    compressed_ends
+        .statistics()
+        .set(Stat::IsConstant, (compressed_ends.len() == 1).into());
+    compressed_ends
+        .statistics()
+        .set(Stat::IsSorted, true.into());
+    compressed_ends
+        .statistics()
+        .set(Stat::IsStrictSorted, true.into());
+    if !compressed_ends.is_empty() {
+        compressed_ends
+            .statistics()
+            .set(Stat::Min, scalar_at(&compressed_ends, 0).vortex_unwrap());
```

Comment:
```suggestion
            .set(Stat::Min, 0.into());
```

Reply:
this isn't necessarily always 0 if we slice?

Reply:
But we just created this array, in general yes but even for a sliced array RunEnd encoding will reset the offset to 0